### DISCUSSION TITLE
Adopt CSD for Cinnamon Settings

### DISCUSSION THREAD

**Discussion by axel358** [linuxmint/discussions#19] [Ideas]
Sorry for being soo lazy but i already mistakenly explained it here

https://github.com/linuxmint/cinnamon-control-center/issues/312

**Comment by Secret-chest** [comment]
In this case I suppose it makes sense, a search bar and a back button. But please don't turn menu bars into header bars.

**Comment by clefebvre** [comment]
There are no menubars in system settings or any of its modules. UI-wise I like the idea. Careful though, if this is done it needs to be done with care...

- In verbose languages (Russian for instance) the title can be quite long. You don't want the search entry to be tiny or the title to be cut because of it. We've got three titlebar buttons to account for, whereas elementary only seems to have one on each side.
- Some of our modules embed their stack switcher into the toolbar, in elementary that's just part of the content pane apparently.
- Modules can be accessed individually (no navigation/search) or via the system settings (back button / search).

Feel free to start a PR for this if you manage to come up with a nice solution but make sure to account for these.

**Comment by Secret-chest** [comment]
I suppose we won't have the stack switcher in the headerbar, just back button, title and a search bar.

**Comment by JosephMcc** [comment]
I wouldn't do this if you can't keep the stack switcher in the headerbar. Putting it in the content pane is a huge waste of vertical space. Certainly not an improvement. We already have some settings windows that are quite tall like mouse/touchpad. This will just make them worse for no real gain. It feels a bit like a do it just because, not for any real reason or improvement.

**Comment by axel358** [comment]
It was just an idea. An idea that didn't take the fact that we can't embed the stack switcher on the headerbar's title area, my bad. Nobody is making any demands
